hack/lib/util/sync.sh

```diff
@@ -0,0 +1,110 @@
+#  - 5 - the destination repository (fork) where changes are targeted
+#  - 6 - the destination repository branch
+# Returns:
+#  None
+function os::sync::fork() {
+    local tmpDir="$1"
+    local fromRepo="$2"
+    local fromBranch="$3"
+    local fromDir="$4"
+    local toRepo="$5"
+    local toBranch="$6"
+    local fromRepoName=$(basename ${fromRepo})
+    local toRepoName=$(basename ${toRepo})
+
+    fromRepoLocation="${tmpDir}/${fromRepoName}"
+    toRepoLocation="${tmpDir}/${toRepoName}"
+
+    mkdir -p "${fromRepoLocation}"
+    mkdir -p "${toRepoLocation}"
+
+    pushd "${fromRepoLocation}" > /dev/null
+    # if we already have a git repo, assume it's the right one and get the requested branch
+    if ! git rev-parse --quiet --is-inside-work-tree &> /dev/null; then
+        git clone --quiet --single-branch --branch="${fromBranch}" "${fromRepo}" "${fromRepoLocation}"
+    fi
+    git fetch --refmap -- origin refs/heads/"${fromBranch}":refs/remotes/origin/"${fromBranch}"
+    git checkout --quiet "${fromBranch}"
+    git reset --quiet --hard origin/"${fromBranch}"
+    firstFromSHA=$(git rev-list --max-parents=0 HEAD)
+    newFromSHA=$(git log --oneline --format='%H' -1)
+    popd > /dev/null
+
+    pushd "${toRepoLocation}" > /dev/null
+    # if we aren't in a git repo, clone it
+    if ! git rev-parse --quiet --is-inside-work-tree &> /dev/null; then
+        git clone --quiet --single-branch --branch="${toBranch}" "${toRepo}" "${toRepoLocation}"
+    fi
+    git fetch --refmap -- origin refs/heads/"${toBranch}":refs/remotes/origin/"${toBranch}"
+    git checkout --quiet "${toBranch}"
+    git reset --quiet --hard origin/"${toBranch}"
+    startingFromSHA="${firstFromSHA}"
+    if [ -f "${fromRepoName}.sha" ]; then
+        startingFromSHA=$(cat "${fromRepoName}.sha")
+    fi
+    if [ "${newFromSHA}" == "${startingFromSHA}" ]; then
+        popd > /dev/null
+        echo "already at level: ${newFromSHA}, nothing to sync"
+        exit 0
+    fi
+
+    os::log::info ""
+    os::log::info "Syncing from ${startingFromSHA}..${newFromSHA}"
+
+    # create a clean branch to start from
+    set +o errexit
+    git branch --quiet -D "${toBranch}-working" &> /dev/null
+    set -o errexit
+    git checkout --quiet -b "${toBranch}-working"
+    popd > /dev/null
+
+    pushd "${fromRepoLocation}" > /dev/null
+    patchDir="${tmpDir}/patches"
+    rm -rf "${patchDir}" && mkdir -p "${patchDir}"
+    index=0
+    for commit in $(git log --format='%H' --no-merges --reverse "${startingFromSHA}..${newFromSHA}" -- "${fromDir}"); do
+        git format-patch --quiet --raw --start-number=${index} --relative="${fromDir}" "${commit}^..${commit}" -o "${patchDir}"
+        index=$((index+=10))
+    done
+    # remove all commits that had no entries
+    find "${patchDir}" -type f -size 0 -exec rm {} \;
+    popd > /dev/null
+
+    pushd "${toRepoLocation}" > /dev/null
+    # only patch if we have patches
+    if [ "$(ls -A ${patchDir})" ]; then
```

Comment:
What are you trying to check here because right now you are checking implicitly that the string returned by the subshell is not empty. If you can't find a better way of doing the check, at least doing it explicitly (`[[ -n ]]`)